gosec: very slow execution after updating to 2.11.1

### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Brew

### Description of the problem

After updating to 2.11.1, lint now takes an hour and a half to run in Github actions. Updating locally from 2.10.0 -> 2.11.1 produces the same result, very very slow execution.

log of running in actions (file paths, etc scrubbed):

<details>

```console
2026-03-06T17:09:58.5618221Z ##[group]run golangci-lint
2026-03-06T17:09:58.5624423Z Running [/github/home/golangci-lint-2.11.1-linux-amd64/golangci-lint config path] in [/__w/project/project] ...
2026-03-06T17:09:58.7075107Z Running [/github/home/golangci-lint-2.11.1-linux-amd64/golangci-lint config verify] in [/__w/project/project] ...
2026-03-06T17:09:59.0492008Z Running [/github/home/golangci-lint-2.11.1-linux-amd64/golangci-lint run] in [/__w/project/project] ...
2026-03-06T18:33:59.0770907Z project/file.go:100:31: G122: Filesystem operation in filepath.Walk/WalkDir callback uses race-prone path; consider root-scoped APIs (e.g. os.Root) to prevent symlink TOCTOU traversal (gosec)
2026-03-06T18:33:59.0772823Z 		contents, err := os.ReadFile(path)
2026-03-06T18:33:59.0773298Z 		                            ^
2026-03-06T18:33:59.0774684Z project/file.go:100:35: G122: Filesystem operation in filepath.Walk/WalkDir callback uses race-prone path; consider root-scoped APIs (e.g. os.Root) to prevent symlink TOCTOU traversal (gosec)
2026-03-06T18:33:59.0776194Z 		fileContents, err := os.ReadFile(path)
2026-03-06T18:33:59.0776668Z 		                                ^
2026-03-06T18:33:59.0777685Z project/file.go:100:20: G703: Path traversal via taint analysis (gosec)
2026-03-06T18:33:59.0778601Z 	err = os.WriteFile(outputOASPath, output, 0600)
2026-03-06T18:33:59.0779550Z 	                  ^
2026-03-06T18:33:59.0780257Z project/file.go:100:38: G704: SSRF via taint analysis (gosec)
2026-03-06T18:33:59.0781304Z 		r, err = http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
2026-03-06T18:33:59.0781889Z 		                                   ^
2026-03-06T18:33:59.0783175Z project/file.go:100:35: G118: context cancellation function returned by WithCancel/WithTimeout/WithDeadline is not called (gosec)
2026-03-06T18:33:59.0784502Z 	ctx, cancel = context.WithTimeout(ctx, 5*time.Minute)
2026-03-06T18:33:59.0784985Z 	                                 ^
2026-03-06T18:33:59.0786696Z project/file.go:100:23: G122: Filesystem operation in filepath.Walk/WalkDir callback uses race-prone path; consider root-scoped APIs (e.g. os.Root) to prevent symlink TOCTOU traversal (gosec)
2026-03-06T18:33:59.0788374Z 		file, err := os.Open(path)
2026-03-06T18:33:59.0788761Z 		                    ^
2026-03-06T18:33:59.0789696Z project/file.go:100:42: G118: context cancellation function returned by WithCancel/WithTimeout/WithDeadline is not called (gosec)
2026-03-06T18:33:59.0791045Z 	revertCtx, cancel := context.WithTimeout(context.Background(), duration+updateConfigTimeout)
2026-03-06T18:33:59.0791849Z 	                                        ^
2026-03-06T18:33:59.0792911Z project/file.go:100:44: G118: context cancellation function returned by WithCancel/WithTimeout/WithDeadline is not called (gosec)
2026-03-06T18:33:59.0794066Z 	lifecycleCtx, cancel := context.WithCancel(ctx)
2026-03-06T18:33:59.0794545Z 	                                          ^
2026-03-06T18:33:59.0795068Z project/file.go:100:40: G704: SSRF via taint analysis (gosec)
2026-03-06T18:33:59.0795979Z 	req, err := http.NewRequestWithContext(ctx, "POST", evaluateURL, bytes.NewBuffer(payloadBytes))
2026-03-06T18:33:59.0796834Z 	                                      ^
2026-03-06T18:33:59.0797538Z 9 issues:
2026-03-06T18:33:59.0797918Z * gosec: 9
2026-03-06T18:33:59.0798072Z 
2026-03-06T18:33:59.0818954Z ##[error]issues found
2026-03-06T18:33:59.0828242Z Ran golangci-lint in 5040028ms
2026-03-06T18:33:59.0828905Z ##[endgroup]
```

</details>

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 2.11.1 built with go1.26.1 from 89a46a2 on 2026-03-06T13:59:43Z
```

</details>

### Configuration

<details>

```yml
# yaml-language-server: $schema=https://golangci-lint.run/jsonschema/golangci.jsonschema.json

version: "2"
linters:
  default: none
  enable:
    - asciicheck
    - bodyclose
    - copyloopvar
    - dogsled
    - errcheck
    - exhaustive
    - goheader
    - gosec
    - govet
    - ineffassign
    - misspell
    - nakedret
    - nestif
    - noctx
    - rowserrcheck
    - sqlclosecheck
    - staticcheck
    - unconvert
    - unparam
    - unused
    - whitespace
  settings:
    exhaustive:
      # Presence of "default" case in switch statements satisfies exhaustiveness,
      # even if all enum members are not listed.
      # Default: false
      default-signifies-exhaustive: true
    misspell:
      locale: US
    nestif:
      # Minimal complexity of if statements to report.
      # Default: 5
      min-complexity: 6
  exclusions:
    generated: lax
    presets:
      - comments
      - common-false-positives
      - legacy
      - std-error-handling
    rules:
      - linters:
          - bodyclose
          - gosec
          - noctx
          - unparam
        path: _test\.go
      - linters:
          - gosec
        path: fixtures.go
      - path: (.+)\.go$
        text: composite literal uses unkeyed fields
      # utils/http error functions (New*f) support %w via custom implementation in NewHttpErrorf,
      # but govet doesn't recognize this pattern
      - linters:
          - govet
        text: "company/project/utils/http\\.New.*f does not support error-wrapping directive %w"
    paths:
      - project/paths
      - third_party$
      - builtin$
      - examples$
formatters:
  enable:
    - gci
    - gofmt
  settings:
    gci:
      # Section configuration to compare against.
      # Section names are case-insensitive and may contain parameters in ().
      # The default order of sections is `standard > default > custom > blank > dot > alias > localmodule`,
      # If `custom-order` is `true`, it follows the order of `sections` option.
      # Default: ["standard", "default"]
      sections:
        - standard # Standard section: captures all standard packages.
        - default # Default section: contains all imports that could not be matched to another section type.
        # - prefix(company/project) # Custom section: groups all imports with the specified Prefix.
        # - blank # Blank section: contains all blank imports. This section is not present unless explicitly enabled.
        # - dot # Dot section: contains all dot imports. This section is not present unless explicitly enabled.
        # - alias # Alias section: contains all alias imports. This section is not present unless explicitly enabled.
        - localmodule # Local module section: contains all local packages. This section is not present unless explicitly enabled.
        # Enable custom order of sections.
        # If `true`, make the section order the same as the order of `sections`.
        # Default: false
      custom-order: true
      # Drops lexical ordering for custom sections.
      # Default: false
      no-lex-order: true
  exclusions:
    generated: lax
    paths:
      - sdk/engine
      - sdk/management
      - sdk/private
      - apis/engine
      - apis/management
      - apis/private
      - third_party$
      - builtin$
      - examples$
issues:
  # after fixing all the issues we don't need to look at just new issues
  new: false

```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.26.1 darwin/arm64
AR='ar'
CC='cc'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='c++'
GCCGO='gccgo'
GO111MODULE=''
GOARCH='arm64'
GOARM64='v8.0'
GOAUTH='netrc'
GOBIN=''
GOCACHE='/Users/user/Library/Caches/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/Users/user/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/8x/f440gk2561j7n3vydj649bgw0000gn/T/go-build4284130322=/tmp/go-build -gno-record-gcc-switches -fno-common'
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMOD='/Users/user/Documents/Projects/company/project/go.mod'
GOMODCACHE='/Users/user/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/user/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/opt/homebrew/Cellar/go/1.26.1/libexec'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/user/Library/Application Support/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/opt/homebrew/Cellar/go/1.26.1/libexec/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.26.1'
GOWORK=''
PKG_CONFIG='pkg-config'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean && golangci-lint run -v
level=info msg="golangci-lint has version 2.11.1 built with go1.26.1 from 89a46a2 on 2026-03-06T13:59:43Z"
level=info msg="[config_reader] Config search paths: [./ /Users/[username]/Documents/Projects/[org]/[project]/main /Users/[username]/Documents/Projects/[org]/[project] /Users/[username]/Documents/Projects/[org] /Users/[username]/Documents/Projects /Users/[username]/Documents /Users/[username] /Users /]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[config_reader] Module name \"[org]/[project]\""
level=info msg="[goenv] Read go env for 5.9825ms: map[string]string{\"GOCACHE\":\"/Users/[username]/Library/Caches/go-build\", \"GOROOT\":\"/opt/homebrew/Cellar/go/1.26.1/libexec\"}"
level=info msg="[lintersdb] Active 23 linters: [asciicheck bodyclose copyloopvar dogsled errcheck exhaustive gci gofmt goheader gosec govet ineffassign misspell nakedret nestif noctx rowserrcheck sqlclosecheck staticcheck unconvert unparam unused whitespace]"
level=info msg="[loader] Go packages loading at mode 8767 (types_sizes|compiled_files|files|imports|name|deps|exports_file) took 8.218555959s"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 113.462375ms"
level=info msg="[linters_context/goanalysis] analyzers took 1h5m50.598805202s with top 10 stages: gosec: 54m36.561285876s, whitespace: 4m10.885249853s, buildir: 1m42.146737216s, unused: 34.339143996s, exhaustive: 23.141457403s, unconvert: 17.477651199s, directives: 14.92005666s, gci: 13.519523206s, S1008: 13.509191578s, unparam: 10.464988094s"
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"third_party$\""
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"sdk/engine\""
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"sdk/management\""
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"builtin$\""
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"examples$\""
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"sdk/private\""
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"apis/engine\""
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"apis/management\""
level=info msg="[runner/exclusion_paths] Skipped 0 issues by pattern \"apis/private\""
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"fixtures.go\", Linters: \"gosec\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Text: \"composite literal uses unkeyed fields\", Path: \"(.+)\\\\.go$\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"sdk/engine\", Linters: \"gci, gofmt\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"sdk/private\", Linters: \"gci, gofmt\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"apis/engine\", Linters: \"gci, gofmt\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"apis/management\", Linters: \"gci, gofmt\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"apis/private\", Linters: \"gci, gofmt\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"third_party$\", Linters: \"gci, gofmt\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"_test\\\\.go\", Linters: \"bodyclose, gosec, noctx, unparam\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Text: \"[org]/[project]/utils/http\\\\.New.*f does not support error-wrapping directive %w\", Linters: \"govet\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"sdk/management\", Linters: \"gci, gofmt\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"builtin$\", Linters: \"gci, gofmt\"]"
level=info msg="[runner/exclusion_rules] Skipped 0 issues by rules: [Path: \"examples$\", Linters: \"gci, gofmt\"]"
level=info msg="[runner] Issues before processing: 66435, after processing: 1"
level=info msg="[runner] Processors filtering stat (in/out): filename_unadjuster: 66435/66435, generated_file_filter: 1/1, fixer: 1/1, max_per_file_from_linter: 1/1, max_from_linter: 1/1, path_shortener: 1/1, path_absoluter: 66435/66435, cgo: 66435/66435, invalid_issue: 66435/1, exclusion_paths: 1/1, uniq_by_line: 1/1, max_same_issues: 1/1, sort_results: 1/1, nolint_filter: 1/1, diff: 1/1, severity-rules: 1/1, path_prettifier: 1/1, path_relativity: 1/1, exclusion_rules: 1/1, source_code: 1/1"
level=info msg="[runner] processing took 3.390754ms with stages: cgo: 1.0505ms, source_code: 1.038708ms, path_absoluter: 489.667µs, filename_unadjuster: 470.001µs, invalid_issue: 195.167µs, exclusion_paths: 66.376µs, exclusion_rules: 62.625µs, uniq_by_line: 4.626µs, sort_results: 4.291µs, path_relativity: 2.751µs, max_same_issues: 1.833µs, generated_file_filter: 1.25µs, path_shortener: 916ns, nolint_filter: 625ns, max_from_linter: 416ns, fixer: 375ns, diff: 251ns, max_per_file_from_linter: 167ns, path_prettifier: 167ns, severity-rules: 42ns"
level=info msg="[runner] linters took 54m27.405131625s with stages: goanalysis_metalinter: 54m27.401593042s"
[path/to/file].go:196:13: operation.OutputIntegrationId undefined (type *"[org]/[project]/apis/engine/v1".Operation has no field or method OutputIntegrationId) (typecheck)
		operation.OutputIntegrationId = &t.OutputIntegrationId
		          ^
1 issues:
* typecheck: 1
level=info msg="File cache stats: 1 entries of total size 6.7KiB"
level=info msg="Memory: 32751 samples, avg is 5428.1MB, max is 5469.2MB"
level=info msg="Execution took 54m35.760525917s"

```

</details>

### A minimal reproducible example or link to a public repository

<details>

simply running the linter. For us this is typically invoked via make, which runs `golangci-lint run --max-same-issues=0 --max-issues-per-linter=0`

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a [sponsor/backer](https://donate.golangci.org) of this project.

--- a/go.mod
+++ b/go.mod
@@ -112,7 +112,7 @@ require (
 	github.com/santhosh-tekuri/jsonschema/v6 v6.0.2
 	github.com/sashamelentyev/interfacebloat v1.1.0
 	github.com/sashamelentyev/usestdlibvars v1.29.0
-	github.com/securego/gosec/v2 v2.24.7
+	github.com/securego/gosec/v2 v2.24.8-0.20260309165252-619ce2117e08
 	github.com/shirou/gopsutil/v4 v4.26.2
 	github.com/sirupsen/logrus v1.9.4
 	github.com/sivchari/containedctx v1.0.3
@@ -146,7 +146,7 @@ require (
 	go.yaml.in/yaml/v3 v3.0.4
 	golang.org/x/exp v0.0.0-20250620022241-b7579e27df2b
 	golang.org/x/mod v0.33.0
-	golang.org/x/sync v0.19.0
+	golang.org/x/sync v0.20.0
 	golang.org/x/sys v0.41.0
 	golang.org/x/tools v0.42.0
 	honnef.co/go/tools v0.7.0

